src/services/download/mod.rs

```diff
@@ -142,15 +138,16 @@ impl DownloadService {
         let hub = Hub::current();
         let slf = self.clone();
 
-        let spawn_result = self
-            .worker
-            .spawn("service.download", Duration::from_secs(300), || {
-                slf.dispatch_download(source, destination).bind_hub(hub)
-            })
-            .await;
+        let _guard = self.worker.enter();
```

Comment:
note to self: read up on why this is needed

Reply:
Good point, I should add a comment on this. The timeout we create below requires an active runtime since it's based on `Sleep`, which internally stores a reference to the runtime's clock. `enter` registers the runtime in thread locals so that we can appropriately create the futures and then spawn them. 

This will go away as soon as we move the executor up and run the entire downloader in the tokio 1 runtime (read: "soon" :) )